packages/labs/analyzer/src/test/types_test.ts

```diff
@@ -0,0 +1,231 @@
+  const type = typeForVariable(module, 'jsdocLocalClass');
+  assert.equal(type.text, 'LocalClass');
+  assert.equal(type.references.length, 1);
+  assert.equal(type.references[0].name, 'LocalClass');
+  assert.equal(type.references[0].package, '@lit-internal/test-types');
+  assert.equal(type.references[0].module, 'module.js');
+  assert.equal(type.references[0].isGlobal, false);
+});
+
+test('jsdocImportedClass', ({module}) => {
+  const type = typeForVariable(module, 'jsdocImportedClass');
+  assert.equal(type.text, 'ImportedClass');
+  assert.equal(type.references.length, 1);
+  assert.equal(type.references[0].name, 'ImportedClass');
+  assert.equal(type.references[0].package, '@lit-internal/test-types');
+  assert.equal(type.references[0].module, 'external.js');
+  assert.equal(type.references[0].isGlobal, false);
+});
+
+test('jsdocStringExternalClassUnion', ({module}) => {
+  const type = typeForVariable(module, 'jsdocStringExternalClassUnion');
+  assert.equal(type.text, 'string | ImportedClass');
+  assert.equal(type.references.length, 1);
+  assert.equal(type.references[0].name, 'ImportedClass');
+  assert.equal(type.references[0].package, '@lit-internal/test-types');
+  assert.equal(type.references[0].module, 'external.js');
+  assert.equal(type.references[0].isGlobal, false);
+});
+
+test('jsdocStringExternalGlobalClassUnion', ({module}) => {
+  const type = typeForVariable(module, 'jsdocStringExternalGlobalClassUnion');
+  assert.equal(type.text, 'string | ImportedClass | HTMLElement');
+  assert.equal(type.references.length, 2);
+  assert.equal(type.references[0].name, 'ImportedClass');
+  assert.equal(type.references[0].package, '@lit-internal/test-types');
+  assert.equal(type.references[0].module, 'external.js');
+  assert.equal(type.references[0].isGlobal, false);
+  assert.equal(type.references[1].name, 'HTMLElement');
+  assert.equal(type.references[1].isGlobal, true);
+});
+
+test('complexType', ({module}) => {
+  const type = typeForVariable(module, 'complexType');
+  assert.equal(type.text, 'Promise<Map<keyof LitElement, ImportedClass[]>>[]');
+  assert.equal(type.references.length, 4);
+  assert.equal(type.references[0].name, 'Promise');
+  assert.equal(type.references[0].isGlobal, true);
+  assert.equal(type.references[1].name, 'Map');
+  assert.equal(type.references[1].isGlobal, true);
+  assert.equal(type.references[2].name, 'LitElement');
+  assert.equal(type.references[2].package, 'lit');
+  assert.equal(type.references[2].module, '');
+  assert.equal(type.references[2].isGlobal, false);
+  assert.equal(type.references[3].name, 'ImportedClass');
+  assert.equal(type.references[3].package, '@lit-internal/test-types');
+  assert.equal(type.references[3].module, 'external.js');
+  assert.equal(type.references[3].isGlobal, false);
+});
+
+test('getImportStringForReferences', ({module}) => {
+  const type = typeForVariable(module, 'complexType');
+  assert.equal(
+    type.getImportStatementsForReferences(),
```

Comment:
This is great. As noted offline, we probably want to collapse and coalesce the import statements in the wrappers. That seems like something we can do as a future improvement.

Reply:
+1, will leave a TODO; we should probably do that as part of actually integrating it into the fw wrappers.

Reply:
Went ahead and added a deduping/coalescing function.